lua/luasnip/extras/postfix.lua

```diff
@@ -0,0 +1,75 @@
+local snip = require("luasnip").snippet
+local events = require("luasnip.util.events")
+local matches = {
+	default = [[[%w%.%_%-%"%']+$]],
+	line = "^.+$",
+}
+
+local function generate_opts(match_pattern, user_callback)
+	return {
+		callbacks = {
+			[-1] = {
+				[events.pre_expand] = function(snippet, event_args)
+					local pos = event_args.expand_pos
+					-- [1]: returns table, gets text end-exclusive.
+					local line_to_cursor = vim.api.nvim_buf_get_text(
+						0,
+						pos[1],
+						0,
+						pos[1],
+						pos[2],
+						{}
+					)[1]
+					local pre_dot = line_to_cursor:match(match_pattern) or ""
+					-- clear pre_dot-text.
+					vim.api.nvim_buf_set_text(
+						0,
+						pos[1],
+						pos[2] - #pre_dot,
+						pos[1],
+						pos[2],
+						{ "" }
+					)
+					snippet.pre_dot = pre_dot
+					if user_callback then
+						user_callback(snippet, event_args)
+					end
+				end,
+			},
+		},
+	}
+end
+
+local function postfix(context, nodes, opts)
+	opts = opts or {}
+	local user_callback = vim.tbl_get(opts, "callbacks", -1, events.pre_expand)
+	vim.validate({
+		context = { context, { "string", "table" } },
+		nodes = { nodes, "table" },
+		opts = { opts, "table" },
+		user_callback = { user_callback, { "nil", "function" } },
+	})
+
+	local match_pattern
+	if type(context) == "string" then
+		context = {
+			trig = context,
+		}
+		match_pattern = matches.default
+	else
+		match_pattern = context.match_pattern or matches.default
+	end
+
+	context = vim.tbl_deep_extend("keep", context, { wordTrig = false })
+	opts = vim.tbl_deep_extend(
+		"force",
+		opts,
+		generate_opts(match_pattern, user_callback)
+	)
```

Comment:
This could be replaced with a single set on `callbacks.-1.pre_expand`, right? Would be a tiny bit clearer IMO

Reply:
Sounds good to me!

Reply:
Is there a nicer way to do this than a couple of nested nil checks?

Reply:
Not really, but those could be moved into a separate function, `luasnip.util.util.tbl_set`,with an interface similar to `vim.tbl_get`, to keep the code readable